TrigScint/python/trigScint.py

```diff
@@ -285,6 +285,42 @@ def __init__(self,name) :
 
 trigScintTrack = TrigScintTrackProducer( "trigScintTrack" )
 
+class TrigScintTrackProducer(ldmxcfg.Producer) :
+    """Configuration for track producer for Trigger Scintillators"""
+
+    def __init__(self,name) :
+        super().__init__(name,'trigscint::','TrigScint')
+
+        self.delta_max = 0.75
+        self.tracking_threshold = 0.  #to add in neighboring channels
+        self.seeding_collection = "TriggerPad1Clusters"
+        self.further_input_collections = ["TriggerPad2Clusters","TriggerPad3Clusters"]
+        self.allow_skip_last_collection = False
+        self.vertical_bar_start_index = 52
+        self.number_horizontal_bars = 24  #16 for x,y segmented geometry only 
+        self.number_vertical_bars = 0     #8 for x,y segmented geometry only
+        self.horizontal_bar_width = 3.
+        self.horizontal_bar_gap = 0.3
+        self.vertical_bar_width = 3.
+        self.vertical_bar_gap = 0.3
+        self.input_pass_name="" #take any pass
+        self.output_collection="TriggerPadTracks"
+        self.verbosity = 0
+
+class TrigScintFirmwareTracker(ldmxcfg.Producer) :
+    """Configuration for the track producer from the Firmware Tracker"""
+    def __init__(self,name) :
+        super().__init__(name,'trigscint::TrigScintFirmwareTracker','TrigScint')
+        self.clustering_threshold=40.0
+        self.digis1_collection='trigScintDigisPad1'
```

Comment:
I will comment the code more thoroughly for sure :)! I am including this code for two reasons. The first is that Elizabeth needs the code for efficiency studies to reflect firmware stuff. The other reason is along the lines of having firmware directly reflect software, and since the firmware uses pattern matching so will the software. This processor does both clustering and tracking in one step, using clusterproducer_sw and trackerproducer_hw to do those steps respectively. These files are the raw ones that are the validated pieces of HLS code, the processor itself just stages the digis. If I wanted to include the hitmaker I could as well, in which case I guess the input would be adc and tdc stuff from QIEDecoder, but I haven't done that just yet.